extmod/uasyncio/tcp.py

```diff
@@ -0,0 +1,76 @@
+# MicroPython uasyncio module
+# MIT license; Copyright (c) 2019-2020 Damien P. George
+
+from . import core
+
+
+# Create a TCP stream connection to a remove host
+async def open_connection(host, port):
+    import usocket as socket
+
+    try:
+        from uerrno import EINPROGRESS
+    except:
+        EINPROGRESS = 115  # Use standard built-in value if uerrno doesn't provide it
+    ai = socket.getaddrinfo(host, port)[0]  # TODO this is blocking!
+    s = socket.socket()
+    s.setblocking(False)
+    ss = core.Stream(s)
+    try:
+        s.connect(ai[-1])
+    except OSError as er:
+        if er.args[0] != EINPROGRESS:
```

Comment:
Should this be `er.args[0] in [EINPROGRESS, EAGAIN]` ?

Reply:
In testing so far I didn't see a need for it to test against EAGAIN.  If you can find a test/situation where this should be EAGAIN then that would be interesting to know.